test_data/proxy/src/com/intellij/memory/agent/proxy/IdeaNativeAgentProxy.java

```diff
@@ -14,17 +14,17 @@ private IdeaNativeAgentProxy() {
 
   public static native boolean canFindPathsToClosestGcRoots();
 
-  public static native long[] getShallowSizeByClasses(Object[] classes);
+  public static native Object getShallowSizeByClasses(Object[] classes, long timeoutInMillis);
 
-  public static native long[] getRetainedSizeByClasses(Object[] classes);
+  public static native Object getRetainedSizeByClasses(Object[] classes, long timeoutInMillis);
 
-  public static native Object getShallowAndRetainedSizeByClasses(Object[] classes);
```

Comment:
Can we use a bit more concrete type? This would easier to notice if we change the return values noticeable.

Reply:
Should we make `MemoryAgentResult` class and bundle it with proxy?

Reply:
Not sure if this gonna be more convenient since you need to create the instance from native code.